velox/expression/fuzzer/SpecialFormSignatureGenerator.h

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/functions/FunctionRegistry.h"
+
+namespace facebook::velox::fuzzer {
+
+class SpecialFormSignatureGenerator {
+ public:
+  virtual ~SpecialFormSignatureGenerator() {}
+
+  /// Generates valid signatures for special forms and append to the signature
+  /// map.
+  void appendSpecialForms(
+      FunctionSignatureMap& signatureMap,
+      const std::string& specialForms) const;
+
+ protected:
+  /// Generates signatures for cast from integral types to the given type and
+  /// adds to input vector.
+  void addCastFromIntegralSignatures(
```

Comment:
Looks like this function is only used in `velox/expression/fuzzer/SpecialFormSignatureGenerator.cpp`, can we move it to anonymous namespace?

Reply:
These functions use the 'makeCastSignature' method, which is needed by both SpecialFormSignatureGenerator and SparkSpecialFormSignatureGenerator, so they cannot be put in the anonymous namespace.